src/modules/ekf2/EKF2.cpp

```diff
@@ -731,6 +731,10 @@ void EKF2::Run()
 		_ekf.setIMUData(imu_sample_new);
 		PublishAttitude(now); // publish attitude immediately (uses quaternion from output predictor)
 
+		// accumulate delta velocity to compute mean acceleration
+		delta_vel_ += _ekf.getVelocityDerivative() * imu_sample_new.delta_vel_dt;
```

Comment:
We could do this directly in the output predictor so that `getVelocityDerivative()` returns the average.

Reply:
Hi @dagar!
Thank you for the review. I have updated the changes, did you mean something like that ?
The drawback would be if someone add a call to `getVelocityDerivative()` somewhere else, the behavior will change here.